geomstats/geometry/lie_algebra.py

```diff
@@ -135,17 +136,66 @@ def matrix_representation(self, basis_representation):
 
         return gs.einsum("...i,ijk ->...jk", basis_representation, self.basis)
 
+    def reshape_metric_matrix(self, metric_matrix):
+        """Reshape diagonal metric matrix to a symmetric matrix of size n.
+
+        Reshape a diagonal metric matrix of size `dim x dim` into a symmetric
+        matrix of size `n x n` where :math: `dim= n (n -1) / 2` is the
+        dimension of the space of skew symmetric matrices. The
+        non-diagonal coefficients in the output matrix correspond to the
+        basis matrices of this space. The diagonal is filled with ones.
+        This useful to compute a matrix inner product.
+
+        Parameters
+        ----------
+        metric_matrix : array-like, shape=[dim, dim]
+            Diagonal metric matrix.
+
+        Returns
+        -------
+        symmetric_matrix : array-like, shape=[n, n]
+            Symmetric matrix.
+        """
+        if Matrices.is_diagonal(metric_matrix):
+            metric_coeffs = gs.diagonal(metric_matrix)
+            metric_mat = gs.abs(
+                self.matrix_representation(metric_coeffs))
+            return metric_mat
+        raise ValueError('This is only possible for a diagonal matrix')
+
     def orthonormal_basis(self, metric_matrix):
```

Comment:
would this be better within the module `riemannian_metric.py`? Having metric-related functions within the Lie algera module could be confusing?

Reply:
Yes, but we need a basis!

Reply:
The basis could be passed in argument?

Reply:
Yes makes sense actually, and to also move reshape metric matrix